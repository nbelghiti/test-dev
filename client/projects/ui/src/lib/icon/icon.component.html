<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
