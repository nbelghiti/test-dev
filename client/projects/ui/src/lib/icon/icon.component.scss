// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
