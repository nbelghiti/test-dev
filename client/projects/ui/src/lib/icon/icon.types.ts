export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
