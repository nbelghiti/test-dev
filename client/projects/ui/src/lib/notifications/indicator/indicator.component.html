<button class="c-notification-indicator">
	<ui-icon name="bell"></ui-icon>

	<span
		*ngIf="count >= 1"
		class="c-notification-indicator__count"
	>
		{{count > 99 ? '99+' : count}}
	</span>
</button>
