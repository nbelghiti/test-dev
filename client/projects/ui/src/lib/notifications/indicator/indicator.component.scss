// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$wrapper-size: 4.4rem;
$indicator-size: 2.4rem;

// =============================================================================
// :: Notification Indicator
// =============================================================================
.c-notification-indicator {
	position: relative;

	display: inline-flex;
	justify-content: center;
	align-items: center;
	width: $wrapper-size;
	height: $wrapper-size;

	font-size: $font-size-base;

	border-radius: 50%;
	border: none;
	background-color: $background-color-grey-96;

	transition:
		background-color $animation-base,
		transform $animation-base;

	cursor: pointer;

	&:hover,
	&:active {
		background-color: $background-color-grey-86;

		transform: scale(0.98) translateY(0.1rem);
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-notification-indicator__count {
	position: absolute;
	top: 0;
	left: 50%;
	z-index: 1;

	width: $indicator-size;
	height: $indicator-size;

	font-weight: $font-weight-ultrabold;

	font-size: $font-size-2x-small;
	line-height: $indicator-size;
	letter-spacing: -0.03rem;
	text-align: center;

	border-radius: 50%;
	background-color: $primary-color;
	color: $white;
}
