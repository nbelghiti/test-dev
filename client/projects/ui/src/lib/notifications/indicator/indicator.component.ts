import { Component, Input } from '@angular/core';

@Component({
	selector: 'ui-notification-indicator',
	templateUrl: './indicator.component.html',
	styleUrls: ['./indicator.component.scss']
})

export class NotificationIndicatorComponent {
	@Input() public count: number;
}
