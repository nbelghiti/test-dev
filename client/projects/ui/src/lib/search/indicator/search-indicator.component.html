<button class="c-search-indicator" (click)="clicked.emit(true)">
	<ui-icon name="search"></ui-icon>
</button>
