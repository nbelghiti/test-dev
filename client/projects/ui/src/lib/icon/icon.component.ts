import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
