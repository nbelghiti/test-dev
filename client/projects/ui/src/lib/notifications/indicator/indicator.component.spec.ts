import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { IconComponent } from '../../icon/icon.component';

import { NotificationIndicatorComponent } from './indicator.component';

describe('NotificationIndicatorComponent', () => {
	let component: NotificationIndicatorComponent;
	let fixture: ComponentFixture<NotificationIndicatorComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
		declarations: [
			NotificationIndicatorComponent,
			IconComponent,
		]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(NotificationIndicatorComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
