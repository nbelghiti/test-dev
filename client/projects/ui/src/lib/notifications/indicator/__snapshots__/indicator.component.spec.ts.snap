// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationIndicatorComponent Should render correctly 1`] = `
<ui-notification-indicator>
  <button
    class="c-notification-indicator"
  >
    <ui-icon
      class="c-icon"
      name="bell"
      ng-reflect-name="bell"
    >
      <i
        class="c-icon__glyph   fas fa-bell"
      />
    </ui-icon>
    
  </button>
</ui-notification-indicator>
`;
